### ISSUE TITLE
Member of object and object can be undefined

### ISSUE THREAD

**Issue by DrChrispoper** [hpersson/eslint-plugin-jsx-expressions#5]
Thanks for the amazing plugin!

I have a use case, one that made us look into having a rule like this. That doesn't seem to be caught yet.
```
const profile: $Profile | undefined
...
<div>
  {profile?.viewCount && <h2>Hey</h2>}
</div>
```

**Comment by alexfoxy** [comment]
For reference here is a test which should throw an error:
```
    {
      code: [
        'const first = { foo: 0 };',
        "<App>{first?.foo && <Foo/>}</App>",
      ].join("\n"),
      errors: [{ messageId: "conditionErrorFalseyString" }],
      parserOptions: {
        ecmaFeatures: {
          jsx: true,
        },
      },
      output: [
        'const first = { foo: 0 };',
        "<App>{!!first?.foo && <Foo/>}</App>",
      ].join("\n"),
      filename: "react.tsx",
    },
```

**Comment by hpersson** [comment]
Thanks for reporting. This should be fixed in the latest release, 1.3.1, which has just been published.
